src/mpyl/project_execution.py

```diff
@@ -1,21 +1,33 @@
 """This module contains the ProjectExecution class."""
-import uuid
+
 from dataclasses import dataclass
+from typing import Optional
 
 from .project import Project
 
 
 @dataclass(frozen=True)
 class ProjectExecution:
     project: Project
-    cache_key: str
+    hashed_changes: Optional[str]
     cached: bool
 
     @staticmethod
-    def always_run(project: Project):
-        """Create a ProjectExecution for a project that will always run, regardless of caching"""
+    def create(project: Project, cached: bool, hashed_changes: Optional[str] = None):
```

Comment:
nit: this structure feels a little bit weird, I guess we could create a factory pattern but then again that also doesn't really fit with Python maybe

Reply:
I do agree, that is kinda what I meant with [this comment](https://github.com/Vandebron/mpyl/pull/410/files#r1571931240). I can take a shot at making this a bit better in a future PR 🙂